OptimoveSDK/Sources/Classes/Optimove.swift

```diff
@@ -52,16 +52,24 @@ typealias Logger = OptimoveCore.Logger
 
         if config.isOptimobileConfigured(), let optimobileConfig = config.optimobileConfig {
             shared.container.resolve { serviceLocator in
-                guard let visitorId = try? serviceLocator.storage().getInitialVisitorId() else {
-                    return
+                do {
+                    let visitorId = try serviceLocator.storage().getInitialVisitorId()
+                    let userId = try? serviceLocator.storage().getCustomerID()
+                    
+                    Optimobile.initialize(config: optimobileConfig, initialVisitorId: visitorId, initialUserId: userId)
+                } catch {
+                    throw GuardError.custom("Failed on OptimobileSDK initialization. Reason: \(error.localizedDescription)")
                 }
-
-                let userId = try? serviceLocator.storage().getCustomerID()
-
-                Optimobile.initialize(config: optimobileConfig, initialVisitorId: visitorId, initialUserId: userId)
             }
         }
     }
+
+    /// Set the credentials for the Optimove server. Intent to use as a step for the delayed initialization.
+    public static func setCredentials(optimoveCredentials: String?, optimobileCredentials: String?) {
+        let builder = OptimoveConfigBuilder(optimoveCredentials: optimoveCredentials, optimobileCredentials: optimobileCredentials)
+        let config = builder.build()
+        initialize(with: config)
```

Comment:
is it true that  Optimobile.initialize is called twice (here and isOptimobileConfigured is always true)? This means

1. 2nd time wont succeed as there is a check if instance is already created
2. some things like sendDeviceInformation, maybeTrackPushDismissedEvents would be also called twice -- would get double event. Also some other helpers etc init twice, can it break smth?

Reply:
Change the initialization logic of `Optimobile`. Now it consumes the config with feature flags and can differentiate inits.

Reply:
it uses new builder here. Does it mean all initial configuration options (like in-app strategy) are lost when we give credentials?